src/components/App.js

```diff
@@ -3,20 +3,143 @@ import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 import calculate from '../logic/calculate';
 
-function App() {
-  const data = { total: '20', next: '5', operation: '+' };
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      decimal: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
```

Comment:
- [ ] The project does not require a `decimal` option for the constructor. This only adds unnecessary complexity to the calculation. Please see the example of [my own solution](https://github.com/voscarmv/calculator/blob/development/src/components/App.js#:~:text=extends%20React.Component%20%7B-,constructor(props)%20%7B,%7D,-render()%20%7B) to this project in order to see how you may simplify this part.